scheme/common/cca/platform/cca_ssd_extractor.go

```diff
@@ -20,57 +20,58 @@ type CcaSsdExtractor struct {
 func (o CcaSsdExtractor) RefValExtractor(rvs comid.ValueTriples) ([]*handler.Endorsement, error) {
 	refVals := make([]*handler.Endorsement, 0, len(rvs.Values))
 
-	for i, rv := range rvs.Values {
+	for _, rv := range rvs.Values {
 		var classAttrs platform.ClassAttributes
 		var refVal *handler.Endorsement
 		var err error
 
 		if err := classAttrs.FromEnvironment(rv.Environment); err != nil {
 			return nil, fmt.Errorf("could not extract PSA class attributes: %w", err)
 		}
+		for i, m := range rv.Measurements.Values {
+			if m.Key == nil {
+				return nil, fmt.Errorf("measurement key is not set at index %d ", i)
+			}
 
-		if rv.Measurement.Key == nil {
```

Comment:
Should we keep this nil test?

I'm unsure why this test was added initially. But if we perform the m.Key.IsSet() test below when the Key is not present, the service will crash.

Reply:
Never mind, I can see the nil test below. Sorry for the noise.